### PR TITLE
TestFlinkRunner calls waitUntilFinish()

### PR DESCRIPTION
**Pull request by jkff**
R: @tgroh 

Will file JIRA now.

### CODE REVIEW

**Comment by jkff** [comment]
Run Flink RunnableOnService

**Comment by jkff** [comment]
Uhhhhh https://github.com/apache/beam/blob/master/runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkRunnerResult.java#L86

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/TestFlinkRunner.java [hunk 1]
```diff
@@ -55,7 +57,18 @@ public static TestFlinkRunner create(boolean streaming) {
   @Override
   public PipelineResult run(Pipeline pipeline) {
     try {
-      return delegate.run(pipeline);
+      PipelineResult result = delegate.run(pipeline);
+      // For tests that don't call waitUntilFinish().
+      try {
+        result.waitUntilFinish(Duration.standardSeconds(30));
```
Comment (tgroh) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/TestFlinkRunner.java]: Why this duration?

Also the nested tries and catches and exception propagation make my head hurt a little. I'm not sure off the top of my head if there's any other way to do it.

You should also consider only calling `cancel` if the job has not already terminated.

Also it turns out the FlinkRunner doesn't support `cancel`

**Comment by jkff** [comment]
This PR is irrelevant cause Flink internally waits for the pipeline to finish; closing.

**Pull request closed without merging**